Linear_cell_complex/include/CGAL/Linear_cell_complex_constructors.h

```diff
@@ -28,65 +29,41 @@ namespace CGAL {
-   * @return A dart created during the convertion.
+
+  /**
+   * Imports a plane-embedded graph from a list of points and edges represented as pairs of vertex indices
    */
   template< class LCC >
   typename LCC::Dart_handle import_from_plane_graph(LCC& alcc,
-                                                    std::istream& ais)
+                                                   const std::vector<typename LCC::Point>& vertices,
+                                                   const std::vector<size_t>& edge_indices)
   {
-    CGAL_static_assertion( LCC::dimension>=2 && LCC::ambient_dimension==2 );
-
+    typedef typename LCC::Traits::Construct_direction_2 Construct_direction_2;
+    typedef typename LCC::Traits::Construct_vector Construct_vector;
     typedef typename LCC::Dart_handle Dart_handle;
     typedef typename LCC::Traits::Direction_2 Direction;
-    typedef typename std::list<Dart_handle>::iterator List_iterator;
     typedef typename std::map<Direction, Dart_handle>::iterator LCC_iterator;
+    typedef typename std::list<Dart_handle>::iterator List_iterator;
+    typedef typename LCC::Point Point;
 
-    // Arrays of vertices
-    std::vector< typename LCC::Vertex_attribute_handle > initVertices;
-    std::vector< std::list<Dart_handle> > testVertices;
-
-    std::string txt;
-    typename LCC::FT x, y;
-    Dart_handle d1=alcc.null_handle;
-    unsigned int v1, v2;
-
-    unsigned int nbSommets = 0;
-    unsigned int nbAretes = 0;
-
-    ais >> nbSommets >> nbAretes;
-    while (nbSommets > 0)
-    {
-      if (!ais.good())
-      {
-        std::cout << "Problem: file does not contain enough vertices."
-                  << std::endl;
-        return alcc.null_handle;
-      }
-
-      ais >> iformat(x) >> iformat(y);
-      initVertices.push_back(alcc.create_vertex_attribute
-                             (typename LCC::Point(x, y)));
-      testVertices.push_back(std::list<Dart_handle>());
-      --nbSommets;
-    }
-
-    while (nbAretes>0)
-    {
-      if (!ais.good())
-      {
-        std::cout << "Problem: file does not contain enough edges."
-                  << std::endl;
-        return alcc.null_handle;
-      }
-
-      // We read an egde (given by the number of its two vertices).
-      ais >> v1 >> v2;
-      --nbAretes;
+    CGAL_static_assertion( LCC::dimension>=2 && LCC::ambient_dimension==2 );
+    CGAL_assertion(edge_indices.size() % 2 == 0);
 
+    std::vector< typename LCC::Vertex_attribute_handle > initVertices;
+    initVertices.reserve(vertices.size());
+    std::transform(vertices.begin(),
+                   vertices.end(),
+                   std::back_inserter(initVertices),
+                   [&](const Point& point) {
+      return alcc.create_vertex_attribute(point);
+    });
+
+    std::vector< std::list<Dart_handle> > testVertices{vertices.size(), std::list<Dart_handle>()};
+
+    Dart_handle d1 = alcc.null_handle;
+    for (auto i = 0; (i + 1) < edge_indices.size(); i += 2) {
```

Comment:
please use std::size_t for comparison with xxx.size(), else you get this kind of warning:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-90/Linear_cell_complex/TestReport_lrineau_ArchLinux.gz

Reply:
Another possibility: use `0u` for the initializer, to that `i` is `unsigned`:
```c++
for (auto i = 0u; (i + 1) < edge_indices.size(); i += 2) {
```

Reply:
Done